test/api/python/sep_log_api.py

```diff
@@ -0,0 +1,240 @@
+    p2_to_y = slv.mkTerm(kinds.SepPto, p2, y)
+
+    # Heap -- the points-to have to be "starred"!
+    heap = slv.mkTerm(kinds.SepStar, p1_to_x, p2_to_y)
+
+    # Constain "nil" to be something random
+    fix_nil = slv.mkTerm(kinds.Equal, nil, expr_nil_val)
+
+    # Add it all to the solver!
+    slv.assertFormula(x_equal_const)
+    slv.assertFormula(y_gt_x)
+    slv.assertFormula(heap)
+    slv.assertFormula(fix_nil)
+
+    # Incremental is disabled due to using separation logic, so don't query
+    # twice!
+
+    r = (slv.checkSat())
+
+    # If this is UNSAT, we have an issue so bail-out
+    if not r.isSat():
+        return -1
+
+    # Obtain our separation logic terms from the solver
+    heap_expr = slv.getValueSepHeap()
+    nil_expr = slv.getValueSepNil()
+
+    # If the heap is not a separating conjunction, bail-out
+    if (heap_expr.getKind() != kinds.SepStar):
+        return -1
+
+    # If nil is not a direct equality, bail-out
+    if (nil_expr.getKind() != kinds.Equal):
+        return -1
+
+    # Obtain the values for our "pointers"
+    val_for_p1 = slv.getValue(p1)
+    val_for_p2 = slv.getValue(p2)
+
+    # We need to make sure we find both pointers in the heap
+    checked_p1 = False
+    checked_p2 = False
+
+    # Walk all the children
+    for child in heap_expr:
+        # If we don't have a PTO operator, bail-out
+        if (child.getKind() != kinds.SepPto):
+            return -1
+
+        # Find both sides of the PTO operator
+        addr = slv.getValue(child[0])
+        value = slv.getValue(child[1])
+
+        # If the current address is the value for p1
+        if (addr == val_for_p1):
+            checked_p1 = True
+
+            # If it doesn't match the random constant, we have a problem
+            if value != random_constant:
+                return -1
+            continue
+
+        if (addr == val_for_p2):
+            checked_p2 = True
+
+            # Our earlier constraint was that what p2 points to must be *greater*
+            # than the random constant -- if we get a value that is LTE, then
+            # something has gone wrong!
+
+            if int(str(value)) <= int(str(random_constant)):
+                return -1
+            continue
+
+        # We should only have two addresses in heap, so if we haven't hit the
+        # "continue" for p1 or p2, then bail-out
+
+        return -1
+
+    # If we complete the loop and we haven't validated both p1 and p2, then we
+    # have a problem
+    if (not checked_p1 or not checked_p2):
+        return -1
+
+    # We now get our value for what nil is
+    value_for_nil = slv.getValue(nil_expr[1])
+
+    # The value for nil from the solver should be the value we originally tied
+    # nil to
+
+    if (value_for_nil != expr_nil_val):
+        return -1
+
+    # All tests pass!
+    return 0
+
+
+# check that we get an exception when we should
+check_exception = validate_exception()
+
+if check_exception:
+    assert False
```

Comment:
```suggestion
assert validate_exception()
```
Notices my suggestion for changes for `validate_exception()`